agents/supervisor/supervisor.go

```diff
@@ -407,6 +408,43 @@ func (s *Supervisor) startBuiltin(agentID string, builtinAgent *agentpb.SetState
 			close(done)
 		}()
 
+	case agentpb.Type_QAN_MYSQL_SLOWLOG_AGENT:
+		params := &mysqlslowlog.Params{
+			DSN:     builtinAgent.Dsn,
+			AgentID: agentID,
+		}
+		m, err := mysqlslowlog.New(params, l)
+		if err != nil {
+			cancel()
+			return err
+		}
+
+		go pprof.Do(ctx, pprof.Labels("agentID", agentID, "type", agentType), m.Run)
+
+		go func() {
+			defer func() {
+				// recover from panic caused by writing to a closed channel
+				if r := recover(); r != nil {
```

Comment:
There is also no way we will merge this. This must not happen.

Reply:
I caught few times with heavy load (sysbench-tcpc).
Will remove it for now - as rare case.